editor/connections_dialog.cpp

```diff
@@ -1320,32 +1363,25 @@ void ConnectionsDock::update_tree() {
 				String descr;
 				bool found = false;
 
-				HashMap<StringName, HashMap<StringName, String>>::Iterator G = descr_cache.find(base);
+				HashMap<StringName, HashMap<StringName, String>>::ConstIterator G = descr_cache.find(doc_class_name);
 				if (G) {
-					HashMap<StringName, String>::Iterator F = G->value.find(signal_name);
+					HashMap<StringName, String>::ConstIterator F = G->value.find(signal_name);
 					if (F) {
 						found = true;
 						descr = F->value;
 					}
 				}
 
 				if (!found) {
-					DocTools *dd = EditorHelp::get_doc_data();
-					HashMap<String, DocData::ClassDoc>::Iterator F = dd->class_list.find(base);
-					while (F && descr.is_empty()) {
-						for (int i = 0; i < F->value.signals.size(); i++) {
-							if (F->value.signals[i].name == signal_name.operator String()) {
-								descr = DTR(F->value.signals[i].description);
+					if (class_doc) {
+						for (int i = 0; i < class_doc->signals.size(); i++) {
+							if (class_doc->signals[i].name == signal_name.operator String()) {
+								descr = DTR(class_doc->signals[i].description);
 								break;
 							}
 						}
-						if (!F->value.inherits.is_empty()) {
-							F = dd->class_list.find(F->value.inherits);
-						} else {
-							break;
-						}
 					}
-					descr_cache[base][signal_name] = descr;
+					descr_cache[doc_class_name][signal_name] = descr;
```

Comment:
The cache is not invalidated until the editor is restarted. This is good for native classes, but bad for scripts. Perhaps we should not cache script signal descriptions?

We could also move the loop outside to prevent unnecessary iterations (actually, this is not so important since a class usually has few signals).

Reply:
I guess not caching script signal descriptions might be fine, provided that it's not too costly to recompute each time.

Otherwise they could be cached separately, and the script cache entry marked dirty whenever the script changes.